stratum/hal/lib/tai/test/tai_manager_test.cc

```diff
@@ -0,0 +1,138 @@
+
+TEST_F(TAIManagerTest, correct_object_creation_test) {
+  EXPECT_TRUE(manager_->IsObjectValid({{TAI_OBJECT_TYPE_MODULE, 0}}));
+  EXPECT_TRUE(manager_->IsObjectValid(
+      {{TAI_OBJECT_TYPE_MODULE, 0}, {TAI_OBJECT_TYPE_HOSTIF, 0}}));
+  EXPECT_TRUE(manager_->IsObjectValid(
+      {{TAI_OBJECT_TYPE_MODULE, 0}, {TAI_OBJECT_TYPE_HOSTIF, 1}}));
+  EXPECT_TRUE(manager_->IsObjectValid(
+      {{TAI_OBJECT_TYPE_MODULE, 0}, {TAI_OBJECT_TYPE_NETWORKIF, 0}}));
+
+  EXPECT_TRUE(manager_->IsObjectValid({{TAI_OBJECT_TYPE_MODULE, 1}}));
+  EXPECT_TRUE(manager_->IsObjectValid(
+      {{TAI_OBJECT_TYPE_MODULE, 1}, {TAI_OBJECT_TYPE_HOSTIF, 0}}));
+  EXPECT_TRUE(manager_->IsObjectValid(
+      {{TAI_OBJECT_TYPE_MODULE, 1}, {TAI_OBJECT_TYPE_HOSTIF, 1}}));
+  EXPECT_TRUE(manager_->IsObjectValid(
+      {{TAI_OBJECT_TYPE_MODULE, 1}, {TAI_OBJECT_TYPE_NETWORKIF, 0}}));
+
+  EXPECT_TRUE(manager_->IsObjectValid({{TAI_OBJECT_TYPE_MODULE, 2}}));
+  EXPECT_TRUE(manager_->IsObjectValid(
+      {{TAI_OBJECT_TYPE_MODULE, 2}, {TAI_OBJECT_TYPE_HOSTIF, 0}}));
+  EXPECT_TRUE(manager_->IsObjectValid(
+      {{TAI_OBJECT_TYPE_MODULE, 2}, {TAI_OBJECT_TYPE_HOSTIF, 1}}));
+  EXPECT_TRUE(manager_->IsObjectValid(
+      {{TAI_OBJECT_TYPE_MODULE, 2}, {TAI_OBJECT_TYPE_NETWORKIF, 0}}));
+
+  EXPECT_TRUE(manager_->IsObjectValid({{TAI_OBJECT_TYPE_MODULE, 3}}));
+  EXPECT_TRUE(manager_->IsObjectValid(
+      {{TAI_OBJECT_TYPE_MODULE, 3}, {TAI_OBJECT_TYPE_HOSTIF, 0}}));
+  EXPECT_TRUE(manager_->IsObjectValid(
+      {{TAI_OBJECT_TYPE_MODULE, 3}, {TAI_OBJECT_TYPE_HOSTIF, 1}}));
+  EXPECT_TRUE(manager_->IsObjectValid(
+      {{TAI_OBJECT_TYPE_MODULE, 3}, {TAI_OBJECT_TYPE_NETWORKIF, 0}}));
+}
+
+TEST_F(TAIManagerTest, get_set_frequency_value_test) {
+  const uint64 kFrequency = 45;
+  SetRequest_Request set_request;
+  set_request.mutable_port()->mutable_frequency()->set_value(kFrequency);
+  EXPECT_TRUE(TAIManager::IsRequestSupported(set_request));
+
+  auto kStatus = manager_->SetValue(set_request, module_netif_pair);
+  EXPECT_TRUE(kStatus.ok());
+
+  DataRequest::Request request(DataRequest::Request::default_instance());
+
+  request.mutable_frequency();
+  auto valueOrStatus = manager_->GetValue(request, module_netif_pair);
+  EXPECT_TRUE(valueOrStatus.ok());
+  EXPECT_EQ(valueOrStatus.ConsumeValueOrDie().mutable_frequency()->value(),
+            kFrequency);
+}
+
+TEST_F(TAIManagerTest, get_set_output_power_value_test) {
+  SetRequest_Request set_request;
+  const int64 kDigits = 1234;
+  const uint32 kPrecision = 2;
+  set_request.clear_port();
+  set_request.mutable_port()
+      ->mutable_output_power()
+      ->mutable_instant()
+      ->set_digits(kDigits);
+  set_request.mutable_port()
+      ->mutable_output_power()
+      ->mutable_instant()
+      ->set_precision(kPrecision);
+
+  EXPECT_TRUE(TAIManager::IsRequestSupported(set_request));
+
+  auto kStatus = manager_->SetValue(set_request, module_netif_pair);
+  EXPECT_TRUE(kStatus.ok());
```

Comment:
Is there a mock class for the TAI adapter? I tried to trace this code but cannot find mocked value for module 0 interface 0.

Reply:
Same as other tests